internal/compiler-bridge/src/main/scala/xsbt/Console.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package xsbt
+
+import java.io.{ PrintWriter, StringWriter }
+
+import xsbti.{ ConsoleResult, Logger }
+
+import scala.tools.nsc.interpreter.IMain
+import scala.tools.nsc.interpreter.Results.{ Incomplete, Success, Error }
+import scala.tools.nsc.{ GenericRunnerCommand, Settings }
+
+// Ported over from sbt-pamflet.
+class Console(args: Array[String], bootClasspathString: String,
+  classpathString: String, initialCommands: String, cleanupCommands: String,
+  loader: ClassLoader, bindNames: Array[String], bindValues: Array[AnyRef],
+  log: Logger) extends xsbti.ConsoleInterface {
+  lazy val interpreterSettings = MakeSettings.sync(args.toList, { message => log.error(Message(message)) })
+  // we need rt.jar from JDK, so java classpath is required
+  val useJavaCp = "-usejavacp"
+  val compilerSettings = MakeSettings.sync(args :+ useJavaCp, bootClasspathString, classpathString, { message => log.error(Message(message)) })
+  if (!bootClasspathString.isEmpty)
+    compilerSettings.bootclasspath.value = bootClasspathString
+  compilerSettings.classpath.value = classpathString
+  val outWriter: StringWriter = new StringWriter
+  val poutWriter: PrintWriter = new PrintWriter(outWriter)
+
+  // log.info(Message("Starting scala interpreter..."))
```

Comment:
We can remove all these comments, we won't need to do any magic with class loaders in Zinc.